### ISSUE TITLE
Log a warning when an undefined property is used

### ISSUE THREAD

**Issue by KirillOsenkov** [Microsoft/msbuild#2989]
When an undefined property is used, consider logging a warning or at least a message. In Strict Mode, fail the build.

**Comment by pakrym** [comment]
You can make a differentiation between when it's used directly like in `'$(FunProperty)'` or in expression `'$(FunProperty)\foo'

**Comment by KirillOsenkov** [comment]
yeah like we have `Exists` for files and directories, have `Defined` for properties.

**Comment by KirillOsenkov** [comment]
See related https://github.com/microsoft/msbuild/issues/5015

**Comment by rainersigwald** [comment]
This can be enabled with `MSBUILDWARNONUNINITIALIZEDPROPERTY`. I presume it wasn't made default when that was implemented in the Stone Age because of noisiness (but we can always reevaluate).

**Comment by DaveCousineau** [comment]
Could have syntax to assert a non-empty value? (`$(RequiredProperty!)`?)

There might be an existing way to check like `<Error Text="Property Missing" Condition="'$(RequiredProperty)' == ''" />` but that has to go in a target not a property group, and you have to refer to the property by name multiple times which kind of defeats the purpose of the check.
